lib/eventFactory.js

```diff
@@ -282,13 +306,29 @@ class EventFactory extends BaseFactory {
         return pipelineFactory.get(config.pipelineId)
             // Sync pipeline to make sure workflowGraph is generated
             .then((p) => {
+                // Sync pipeline with the parentEvent sha and create jobs based on that sha
                 if (config.parentEventId) {
                     modelConfig.parentEventId = config.parentEventId;
 
-                    // Sync pipeline with the parentEvent sha and create jobs based on that sha
+                    // for child pipelines restart event, sync with configPipelineSha
+                    if (config.configPipelineSha) {
```

Comment:
could add jsdoc for `config.configPipelineSha`

Reply:
Added :)